backend/courses/models.py

```diff
@@ -399,6 +408,35 @@ class Meta:
     def __str__(self):
         return "%s %s" % (self.full_code, self.course.semester)
 
+    @property
+    def current_popularity(self):
+        """
+        The current popularity of the section, which is defined as:
+        [the number of active PCA registrations for this section]/[the class capacity]
+        mapped onto the range [0,1] where the lowest current popularity (across all sections)
+        maps to 0 and the highest current popularity maps to 1.
+        NOTE: sections with an invalid class capacity (0 or negative) are excluded from
+        computation of the statistic, and if this section has an invalid class capacity, then
+        this property will equal None (or null in JSON).
+        """
+        from alert.models import Registration  # imported here to avoid circular imports
+
+        if self.capacity == 0:
+            return None
+        aggregate_scores = (
+            Registration.objects.filter(section__course__semester=get_current_semester(),
```

Comment:
Does this query need to be run every time? Not sure if it's an expensive one, but `aggregate_scores` is the same across all sections, so wondering if there's any way to cache it easily.